WordPress/gutenberg
# [Mobile] - Font size support for the blocks: Button, Preformatted and Heading levels.

- Gutenberg Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/4071

## Description
This PR continues to add support for font size to other pending blocks: `Button`, `Preformatted`, `Code` as well as extend support for the levels (h1, h2, h3, etc.) of the `Heading` block.

- **Button** block: Support for font size.
- **Code** block: Support for font size. (This block is still **behind** the DEV flag).
- **Preformatted** block: Support for color and background styles as well as font size support. If a block-based theme is active, the background of the block will be transparent.
- **Heading** block: Support for font size to the different levels (h1, h2, h3, h4, etc). This is for default font size values coming from the `theme.json` [see this example](https://github.com/Automattic/themes/blob/trunk/quadrat/theme.json#L585-L669).
- **PlainText** component: now parses any fontsize value through `getPxFromCssUnit`.

**NOTE**: On **iOS**, there's a current issue where changing the font size is **not reflected visually** for the following blocks: `Heading`, `Verse` and `Preformatted`. This is currently being investigated and hopefully addressed in another PR.

## How has this been tested?
### Test case 1 - Button block

- Open the app
- Add a `Buttons` block
- **Expect** the button to be focused
- Type some text
- Open the block's settings
- Change the font size
- **Expect** the font size to change accordingly

### Test case 2 - Preformatted (Font size)

- Open the app **(On Android)**
- Add a `Preformatted` block
- Type some text
- Open the block's settings
- Change the font size
- **Expect** the font size to change accordingly

### Test case 3 - Preformatted (Text and background colors)

- Open the app
- Add a `Preformatted` block
- Type some text
- Open the block's settings
- Set a background and text color
- **Expect** the colors to be visible

### Test case 4 - Heading levels

- Open the app (Using WordPress Android with the **Quadrat** theme enabled)
- Add a `Heading` block
- Select the level H1
- **Expect** the block to have the right default font size
- Repeat the same for the following levels: H2, H3

## Screenshots 

Button block |Preformatted
-|-
<kbd><img src="https://user-images.githubusercontent.com/4885740/135826975-6f2241cf-85b4-4c99-a514-038c2adadd63.png" width="200" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/135826982-390fc518-0288-475a-8c8d-75a225368bdc.png" width="200" /></kbd>

Preformatted Android |Heading Android
-|-
<kbd><img src="https://user-images.githubusercontent.com/4885740/135827158-bd6d4d47-dd5a-4caf-882d-9b5f708b3869.png" width="200" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/135827187-cbdd4316-e21c-4de1-ba40-73e52521bac7.png" width="200" /></kbd>

## Types of changes
New feature

## Checklist:
